contracts/foundation/provider_repository.go

```diff
@@ -0,0 +1,22 @@
+package foundation
+
+type ProviderRepository interface {
+	// Boot boots all registered service providers in dependency order.
+	Boot(app Application)
+
+	// GetBooted returns a slice of all providers that have been booted.
+	GetBooted() []ServiceProvider
+
+	// LoadFromConfig lazy-loads providers from the "app.providers" config.
+	LoadFromConfig(app Application) []ServiceProvider
+
+	// Register sorts and registers all configured providers in dependency order.
+	Register(app Application) []ServiceProvider
```

Comment:
The returned value seems to be unnecessary.

```suggestion
	Register()
```